// uglify-js-brunch/lib/index.js
// Generated by CoffeeScript 1.4.0
(function() {
  var UglifyMinifier, sysPath, uglify;

  sysPath = require('path');

  uglify = require('uglify-js');

  module.exports = UglifyMinifier = (function() {

    UglifyMinifier.prototype.brunchPlugin = true;

    UglifyMinifier.prototype.type = 'javascript';

    function UglifyMinifier(config) {
      var _ref, _ref1, _ref2;
      this.config = config;
      this.options = (_ref = (_ref1 = this.config) != null ? (_ref2 = _ref1.plugins) != null ? _ref2.uglify : void 0 : void 0) != null ? _ref : {};
    }

    UglifyMinifier.prototype.minify = function(data, path, callback) {
      var error, minified;
      try {
        minified = uglify.minify(data, {
          fromString: true
        }).code;
      } catch (err) {
        error = "JS minify failed on " + path + ": " + err;
      }
      return process.nextTick(function() {
        return callback(error, minified || data);
      });
    };

    return UglifyMinifier;

  })();

}).call(this);
